extensions/interactions/AlgebraicExpressionInput/directives/algebraic-expression-input-validation.service.spec.ts

```diff
@@ -184,4 +188,52 @@ describe('AlgebraicExpressionInputValidationService', () => {
       customizationArgs, answerGroups, goodDefaultOutcome);
     expect(warnings).toEqual([]);
   });
+
+  it('should warn if there are missing custom variables', function() {
+    answerGroups[0].rules = [
+      rof.createFromBackendDict({
+        rule_type: 'IsEquivalentTo',
+        inputs: {
+          x: 'x^2 + alpha - y/b'
+        }
+      })
+    ];
+    customizationArgs = {
+      customOskLetters: {
+        value: ['y', 'a', 'b']
+      }
+    };
+
+    warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups, goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: AppConstants.WARNING_TYPES.ERROR,
+      message: (
+        'The following variables are present in some of the answer groups ' +
+        'but are missing from the custom letters list: α,x')
```

Comment:
Is this a learner facing error message? If so, how does a learner know what the "custom letters list" is?

Reply:
No, this is a creator facing message and the creator has the ability to modify the custom letters and the rules.